pilot/pkg/networking/core/v1alpha3/loadbalancer/loadbalancer.go

```diff
@@ -136,7 +136,9 @@ func applyLocalityWeight(
 
 			// remove groups of endpoints in a locality that miss matched
 			for i := range misMatched {
-				loadAssignment.Endpoints[i].LbEndpoints = nil
+				if loadAssignment.Endpoints[i] != nil {
```

Comment:
This seems suspicious. How does this happen? is it from a test hitting only low level functions with obscure inputs or can it really be triggered?

Reply:
This one I wasn't too sure of - but I think the culprit is `FuzzApplyLocalityLBSetting` in `pilot/pkg/networking/core/v1alpha3/loadbalancer/fuzz_test.go`.  So I'm assuming that falls under the category of obscure inputs.  All other cases where Endpoints are set, there is no possibility of an endpoint being nil